lib/auth/methods.go

```diff
@@ -382,7 +394,13 @@ func (a *Server) authenticateUserInternal(ctx context.Context, req AuthenticateU
 					Webauthn: wantypes.CredentialAssertionResponseToProto(req.Webauthn),
 				},
 			}
-			requiredExt := &mfav1.ChallengeExtensions{Scope: mfav1.ChallengeScope_CHALLENGE_SCOPE_LOGIN}
+			requiredExt := &mfav1.ChallengeExtensions{Scope: requiredScope}
+			// If the user is trying to change their password and doesn't use their
+			// old password, we need to require identity verification (i.e. a resident
+			// token used for MFA).
+			if requiredScope == mfav1.ChallengeScope_CHALLENGE_SCOPE_CHANGE_PASSWORD && req.Pass == nil {
+				requiredExt.UserVerificationRequirement = string(protocol.VerificationRequired)
```

Comment:
Let's move this to ChangePassword, I think it's better placed in a "higher" layer. `authenticateUserInternal` shouldn't be concerned with the specifics of changing passwords.